ui/ducks/bridge/bridge.ts

```diff
@@ -1,24 +1,66 @@
-import { createSlice } from '@reduxjs/toolkit';
-
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
 import { Hex } from '@metamask/utils';
+import { getAddress } from 'ethers/lib/utils';
 import { swapsSlice } from '../swaps/swaps';
 import { SwapsTokenObject } from '../../../shared/constants/swaps';
 import { SwapsEthToken } from '../../selectors';
+import { fetchTokenExchangeRates } from '../../helpers/utils/util';
+import {
+  QuoteMetadata,
+  QuoteResponse,
+  SortOrder,
+} from '../../pages/bridge/types';
 
 export type BridgeState = {
   toChainId: Hex | null;
   fromToken: SwapsTokenObject | SwapsEthToken | null;
   toToken: SwapsTokenObject | SwapsEthToken | null;
   fromTokenInputValue: string | null;
+  fromTokenExchangeRate: number | null;
+  toTokenExchangeRate: number | null;
+  sortOrder: SortOrder;
+  selectedQuote: (QuoteResponse & QuoteMetadata) | null; // Alternate quote selected by user. When quotes refresh, the best match will be activated.
 };
 
 const initialState: BridgeState = {
   toChainId: null,
   fromToken: null,
   toToken: null,
   fromTokenInputValue: null,
+  fromTokenExchangeRate: null,
+  toTokenExchangeRate: null,
+  sortOrder: SortOrder.COST_ASC,
+  selectedQuote: null,
 };
 
+export const setSrcTokenExchangeRates = createAsyncThunk(
+  'bridge/setSrcTokenExchangeRates',
+  async (request: { chainId: Hex; tokenAddress: string; currency: string }) => {
+    const { chainId, tokenAddress, currency } = request;
+    const exchangeRates = await fetchTokenExchangeRates(
+      currency,
+      [tokenAddress],
+      chainId,
+    );
+    return exchangeRates?.[getAddress(tokenAddress)];
+  },
+);
+
+export const setDestTokenExchangeRates = createAsyncThunk(
+  'bridge/setDestTokenExchangeRates',
+  async (request: { chainId: Hex; tokenAddress: string; currency: string }) => {
+    const { chainId, tokenAddress, currency } = request;
+    const exchangeRates = await fetchTokenExchangeRates(
+      currency,
+      [tokenAddress],
+      chainId,
+    );
+    return {
+      toTokenExchangeRate: exchangeRates?.[getAddress(tokenAddress)],
+    };
+  },
+);
+
```

Comment:
The above two thunks do roughly the same thing. Is there a way we can reuse the logic within function at all? 
I do see that there are different properties and object mappings in the return, but they could be handled after return?

Reply:
nice catch! I refactored these methods separately and missed the return line of `setDestTokenExchangeRates`. They can pretty much use the same logic. I'll clean this up